test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -53,31 +73,57 @@ struct test_inclusive_scan_by_segment
         }
     }
 
-    template <typename Iterator1, typename Iterator2, typename Size>
+#ifdef DUMP_CHECK_RESULTS
+    template <typename Iterator, typename Size>
+    void display_param(const char* msg, Iterator it, Size n)
+    {
+        using DifferenceType = typename DifferenceTypeT<Iterator>::DifferenceType;
+
+        std::cout << msg;
+        for (DifferenceType i = 0; i < n; ++i)
+        {
+            if (i > 0)
+                std::cout << ", ";
+            std::cout << it[i];
+        }
+        std::cout << std::endl;
+    }
+#endif // DUMP_CHECK_RESULTS
+
+    template <typename Iterator1, typename Iterator2, typename Iterator3, typename Size,
+              typename BinaryOperationCheck = oneapi::dpl::__internal::__pstl_plus>
     void
-    check_values(Iterator1 host_keys, Iterator2 val_res, Size n)
+    check_values(Iterator1 host_keys, Iterator2 host_vals, Iterator3 val_res, Size n,
+                 BinaryOperationCheck op = BinaryOperationCheck())
     {
-        //T keys[n1] = { 1, 2, 3, 4, 1, 1, 2, 2, 3, 3, 4, 4, 1, 1, 1, ...};
-        //T vals[n1] = { 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 3, ...};
+        // https://docs.oneapi.io/versions/latest/onedpl/extension_api.html
+        // keys:   [ 0, 0, 0, 1, 1, 1 ]
+        // values: [ 1, 2, 3, 4, 5, 6 ]
+        // result: [ 1, 1 + 2 = 3, 1 + 2 + 3 = 6, 4, 4 + 5 = 9, 4 + 5 + 6 = 15 ]
+
+#ifdef DUMP_CHECK_RESULTS
+        std::cout << "check_values(n = " << n << ") : " << std::endl;
+        display_param("keys:   ", host_keys, n);
+        display_param("values: ", host_vals, n);
+        display_param("result: ", val_res,   n);
+#endif // DUMP_CHECK_RESULTS
+
+        if (n < 1)
+            return;
+
+        using ValT = typename ::std::decay<decltype(val_res[0])>::type;
+
+        std::vector<ValT> expected_val_res(n);
+        inclusive_scan_by_segment_serial(host_keys, host_vals, expected_val_res, n, op);
+
+#ifdef DUMP_CHECK_RESULTS
+        display_param("expected result: ", expected_val_res.data(), n);
+#endif // DUMP_CHECK_RESULTS
 
-        int segment_length = 1;
-        auto expected_segment_sum = 1;
-        auto current_key = host_keys[0];
-        auto current_sum = 0;
-        for (int i = 0; i != n; ++i)
+        using DifferenceType = typename DifferenceTypeT<decltype(host_vals)>::DifferenceType;
+        for (DifferenceType i = 0; i < n; ++i)
```

Comment:
Possible using Size instead of DifferenceType.

Reply:
Fixed